packages/hint-manifest-scoped-urls/src/hint.ts

```diff
@@ -0,0 +1,112 @@
+ */
+
+import { URL } from 'url';
+import { relative } from 'path';
+import { IHint, HintContext, NetworkData, ProblemLocation } from 'hint';
+import { ManifestEvents, ManifestParsed } from '@hint/parser-manifest';
+import { debug as d } from '@hint/utils';
+
+import meta from './meta';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class ManifestScopedUrlsHint implements IHint {
+    public static readonly meta = meta;
+    public constructor(context: HintContext<ManifestEvents>) {
+        /**
+         * See if the `start_url` is accessible.
+         */
+        const startUrlAccessible = async (startUrl: string, resource: string, startUrllocation: ProblemLocation): Promise<boolean> => {
+            let networkData: NetworkData;
+
+            try {
+                networkData = await context.fetchContent(startUrl);
+            } catch (e) {
+                debug(`Failed to fetch ${startUrl}`);
+                const message = `Request failed for 'start_url'`;
+
+                context.report(resource, message, { location: startUrllocation });
+
+                return false;
+            }
+
+            const response = networkData.response;
+
+            if (response.statusCode !== 200) {
+                const message = `Specified 'start_url' is not accessible. Status code: '${response.statusCode}'.`;
+
+                context.report(resource, message, { location: startUrllocation });
+
+                return false;
+            }
+
+            return true;
+        };
+
+        /**
+         * Checks that the `start_url` is under the scope of
+         * the URL specified in the `scope`
+         * @param parsedContent
+         * @param resource
+         */
+        const startUrlInScope = (startURL: string, scope: string, resource: string, startUrllocation: ProblemLocation): boolean => {
+            const relativePath = relative(scope, startURL);
+            const inScope = relativePath && !relativePath.startsWith('..');
+
+            if (relativePath === '') {
+                return true;
+            }
+
+            if (!inScope) {
+                const message = `'start_url' is not in scope of the app.`;
+
+                context.report(resource, message, { location: startUrllocation });
+
+                return false;
+            }
+
+            return true;
+        };
+
+        const validate = async ({ getLocation, parsedContent: { start_url: startURL, scope }, resource }: ManifestParsed) => {
+            const resourceURL = new URL(resource);
+            const hostnameWithProtocol = `${resourceURL.protocol}//${resourceURL.host}`;
+
+            debug(`Validating hint manifest-scoped-urls`);
+
+            if (startURL) {
+                const startUrlLocation = getLocation('start_url');
```

Comment:
Better to show we know this exists at the earliest opportunity.
```suggestion
                const startUrlLocation = getLocation('start_url')!;
```